src/compiler/crystal/macros/interpreter.cr

```diff
@@ -376,25 +388,83 @@ module Crystal
     end
 
     def visit(node : Yield)
+      orig_str = @str
+      @str = IO::Memory.new
+      expand_yield(node)
+
+      @last = MacroId.new(@str.to_s)
+
+      @str = orig_str
```

Comment:
I was thinking if it won't be better to use `String.build`, then passing the io to `expand_yield(node, io)`.

This will prevent to have this temporary assignments.

Reply:
Then what do we do with `io` in `expand_yield`? The IO needs to be available to nested evaluations (via `parsed_body.accept self`). Assigning `@str` is just the way to do that.